src/fhirtypes/ElementDefinition.ts

```diff
@@ -1162,6 +1162,17 @@ export class ElementDefinition {
         break;
       case 'Reference':
         value = value as FshReference;
+        if (value.sdType) {
+          const validTypes: string[] = [];
+          this.type.forEach(t =>
+            t.targetProfile.map(tp => validTypes.push(fisher.fishForMetadata(tp).sdType))
```

Comment:
Although it would be weird, I think it is possible that the `targetProfile` is something that the `fisher` fails to find, if that is the case here, we'd get an undefined error instead of a nicer InvalidTypeError. So I think adding some `?` here could help that. 

Also sorry for this maybe pedantic comment, but wouldn't this line make more sense as a `forEach` than a `map`?

Reply:
I updated the inner loop here to use a `forEach` and to only add types that are found.